lte/gateway/c/core/oai/lib/store/sqlite.cpp

```diff
@@ -0,0 +1,142 @@
+  _create_store();
+}
+
+std::vector<std::string> SqliteStore::_create_db_locations(
+    std::string db_location, int n_shards) {
+  // in memory if db_location is not specified
+  if (db_location.length() == 0) {
+    db_location = "/var/opt/magma/";
+  }
+
+  std::vector<std::string> db_location_list;
+  for (int shard = 0; shard < n_shards; shard++) {
+    std::string to_push = "file:" + db_location + "subscriber" +
+                          std::to_string(shard) + ".db?cache=shared";
+    db_location_list.push_back(to_push);
+    std::cout << "[LOG] DB location: " << db_location_list[shard] << std::endl;
+  }
+  return db_location_list;
+}
+
+void SqliteStore::_create_store() {
+  int rc;
+  for (auto db_location_s : _db_locations) {
+    sqlite3* db;
+    int rc;
+    const char* db_location = db_location_s.c_str();
+    rc = sqlite3_open(db_location, &db);
+    if (rc) {
+      std::cout << "Cannot open database " << sqlite3_errmsg(db) << std::endl;
+    } else {
+      std::cout << "Database opened successfully at" << db_location
+                << std::endl;
+    }
+
+    const char* sql =
+        "CREATE TABLE IF NOT EXISTS subscriberdb"
+        "(subscriber_id text PRIMARY KEY, data text)";
+    char* zErrMsg;
+
+    rc = sqlite3_exec(db, sql, NULL, 0, &zErrMsg);
+    if (rc != SQLITE_OK) {
+      std::cout << "SQL Error " << zErrMsg << std::endl;
+      sqlite3_free(zErrMsg);
+    } else {
+      std::cout << "Table created successfully!!" << std::endl;
+    }
+
+    sqlite3_close(db);
+  }
+}
+
+void SqliteStore::add_subscriber(const SubscriberData& subscriber_data) {
+  std::string sid_s = _to_str(subscriber_data);
+  const char* sid = sid_s.c_str();
+  std::string data_str_s;
+  subscriber_data.SerializeToString(&data_str_s);
+  std::cout << "Serialized subscriber data: " << data_str_s << std::endl;
+  const char* data_str = data_str_s.c_str();
+
+  std::string db_location_s = _db_locations[_sid2bucket(sid)];
+  const char* db_location = db_location_s.c_str();
+  sqlite3* db;
+  int rc_open = sqlite3_open(db_location, &db);
+  if (rc_open) {
+    std::cout << "Cannot open database " << sqlite3_errmsg(db) << std::endl;
+  } else {
+    std::cout << "Database " << db_location << " opened successfully "
+              << std::endl;
+  }
+  const char* sql =
+      "INSERT INTO subscriberdb(subscriber_id, data) "
+      "VALUES (?, ?)";
+  sqlite3_stmt* stmt;
+  const char* pzTail;
+  int rc_prep = sqlite3_prepare_v2(db, sql, strlen(sql), &stmt, &pzTail);
+  if (rc_prep == SQLITE_OK) {
+    sqlite3_bind_text(stmt, 1, sid, strlen(sid), NULL);
+    sqlite3_bind_blob(stmt, 2, data_str, strlen(data_str), NULL);
+    std::cout << "Successful data binding" << std::endl;
+    sqlite3_step(stmt);
+    sqlite3_finalize(stmt);
+    std::cout << "Finalized the sqlite write operation" << std::endl;
+  } else {
+    std::cout << "SQL Error " << std::endl;
+  }
+}
+
+const char* SqliteStore::_to_str(const SubscriberData& subscriber_data) {
```

Comment:
This function is not converting SubscriberData object to string, but the Subscriber ID of the given subscriber. Please rename it to `get_sid` or `get_imsi`.